tests/importers/test_autoconfig.py

```diff
@@ -0,0 +1,155 @@
+import shutil
+from pathlib import Path
+from typing import List, Text, Set
+from unittest.mock import Mock
+
+import pytest
+from _pytest.capture import CaptureFixture
+from _pytest.monkeypatch import MonkeyPatch
+
+from rasa.constants import CONFIG_AUTOCONFIGURABLE_KEYS
+from rasa.importers import autoconfig
+from rasa.utils import io as io_utils
+
+CONFIG_FOLDER = Path("data/test_config")
+
+SOME_CONFIG = CONFIG_FOLDER / "stack_config.yml"
+DEFAULT_CONFIG = Path("rasa/importers/default_config.yml")
+
+
+@pytest.mark.parametrize(
+    "config_path, autoconfig_keys",
+    [
+        (Path("rasa/cli/initial_project/config.yml"), ["pipeline", "policies"]),
+        (CONFIG_FOLDER / "config_policies_empty.yml", ["policies"]),
+        (CONFIG_FOLDER / "config_pipeline_empty.yml", ["pipeline"]),
+        (CONFIG_FOLDER / "config_policies_missing.yml", ["policies"]),
+        (CONFIG_FOLDER / "config_pipeline_missing.yml", ["pipeline"]),
+        (SOME_CONFIG, []),
+    ],
+)
+def test_get_configuration(
+    config_path: Path, autoconfig_keys: Set, monkeypatch: MonkeyPatch
+):
+    def _auto_configure(_, keys_to_configure):
+        return keys_to_configure
+
+    monkeypatch.setattr(autoconfig, "_dump_config", Mock())
+    monkeypatch.setattr(autoconfig, "_auto_configure", _auto_configure)
+
+    config = autoconfig.get_configuration(str(config_path))
+
+    if autoconfig_keys:
+        expected_config = _auto_configure(config, autoconfig_keys)
+    else:
+        expected_config = config
+
+    assert sorted(config) == sorted(expected_config)
+
+
+@pytest.mark.parametrize("config_file", ("non_existent_config.yml", None))
+def test_get_configuration_missing_file(tmp_path: Path, config_file: Text):
+    if config_file:
+        config_file = tmp_path / "non_existent_config.yml"
+
+    config = autoconfig.get_configuration(str(config_file))
+
+    assert config == {}
+
+
+@pytest.mark.parametrize(
+    "keys_to_configure", (["policies"], ["pipeline"], ["policies", "pipeline"])
+)
+def test_auto_configure(keys_to_configure: List[Text]):
+    default_config = io_utils.read_config_file(DEFAULT_CONFIG)
+
+    config = autoconfig._auto_configure({}, keys_to_configure)
+
+    for k in keys_to_configure:
+        assert config[k] == default_config[k]  # given keys are configured correctly
```

Comment:
optional: no need to change it, just wanted to point it out 😆 
```suggestion
    assert all(config[k] == default_config[k] for k in keys_to_configure):
```